src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,97 @@
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {BehaviorSubject, Observable, throwError} from 'rxjs';
+import {Auth} from '../../configuration/interfaces/application-properties';
+import {ConfigurationService} from '../../configuration/services/configuration-service';
+import {AuthRequest} from '../models/auth-request';
+import {AuthResponse} from '../models/auth-response';
+import {catchError, tap} from 'rxjs/operators';
+import {User} from '../models/user';
+import {Router} from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  private ENDPOINT_LOGIN = 'login';
+  private _resource: Auth;
+  public user: BehaviorSubject<User> = new BehaviorSubject<User>(null);
+  private tokenExpirationTimer: any;
+
+  constructor(private _httpClient: HttpClient,
+              private _config: ConfigurationService,
+              private router: Router) {
+    this._resource = this._config.get().providers.auth;
+  }
+
+  public login(username: string, password: string): Observable<AuthResponse> {
+    const request: AuthRequest = {username, password};
+    const key = 'login';
+    return this._httpClient.post<AuthResponse>(
+      this._resource.address + this._resource.endpoints[this.ENDPOINT_LOGIN], request)
```

Comment:
ENDPOINT_LOGIN som tam dal ale asi vieme pristupit aj cez . operator nie? e.g. `this._resource.endpoints.login`